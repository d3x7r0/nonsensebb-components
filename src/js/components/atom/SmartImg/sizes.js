const IMAGE_SIZES = [
  120,
  160,
  240,
  320,
  480,
  768,
  920,
  1280,
  1600,
  1920,
  2560,
  3840,
]

export function calculateSizes({ min, max } = {}) {
  let result = Array.from(IMAGE_SIZES)

  if (min) {
    result = result.filter(entry => min < entry)
  }

  if (max) {
    result = result.filter((entry, idx, arr) => {
      const prev = arr[idx - 1] || 0

      return max >= entry || max >= prev
    })
  }

  return result
}
