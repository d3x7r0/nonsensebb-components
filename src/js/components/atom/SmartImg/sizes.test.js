import { calculateSizes } from './sizes'

describe('components/atom/SmartImg', () => {
  describe('calculateSizes', () => {
    it('handles min value correctly', () => {
      const result1 = calculateSizes({ min: 100 })
      expect(result1).toMatchSnapshot()

      const result2 = calculateSizes({ min: 1000 })
      expect(result2).toMatchSnapshot()

      const result3 = calculateSizes({ min: 4000 })
      expect(result3).toMatchSnapshot()
    })

    it('handles max value correctly', () => {
      const result1 = calculateSizes({ max: 100 })
      expect(result1).toMatchSnapshot()

      const result2 = calculateSizes({ max: 1280 })
      expect(result2).toMatchSnapshot()

      const result3 = calculateSizes({ max: 1800 })
      expect(result3).toMatchSnapshot()

      const result4 = calculateSizes({ max: 4000 })
      expect(result4).toMatchSnapshot()
    })

    it('handles min and max values correctly', () => {
      const result1 = calculateSizes({
        min: 0,
        max: 100,
      })
      expect(result1).toMatchSnapshot()

      const result2 = calculateSizes({
        min: 300,
        max: 1300,
      })
      expect(result2).toMatchSnapshot()

      const result3 = calculateSizes({
        min: 800,
        max: 4000,
      })
      expect(result3).toMatchSnapshot()
    })
  })
})
