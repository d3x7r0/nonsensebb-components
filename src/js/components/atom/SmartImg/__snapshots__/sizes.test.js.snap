// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atom/SmartImg calculateSizes handles max value correctly 1`] = `
Array [
  120,
]
`;

exports[`components/atom/SmartImg calculateSizes handles max value correctly 2`] = `
Array [
  120,
  160,
  240,
  320,
  480,
  768,
  920,
  1280,
  1600,
]
`;

exports[`components/atom/SmartImg calculateSizes handles max value correctly 3`] = `
Array [
  120,
  160,
  240,
  320,
  480,
  768,
  920,
  1280,
  1600,
  1920,
]
`;

exports[`components/atom/SmartImg calculateSizes handles max value correctly 4`] = `
Array [
  120,
  160,
  240,
  320,
  480,
  768,
  920,
  1280,
  1600,
  1920,
  2560,
  3840,
]
`;

exports[`components/atom/SmartImg calculateSizes handles min and max values correctly 1`] = `
Array [
  120,
]
`;

exports[`components/atom/SmartImg calculateSizes handles min and max values correctly 2`] = `
Array [
  320,
  480,
  768,
  920,
  1280,
  1600,
]
`;

exports[`components/atom/SmartImg calculateSizes handles min and max values correctly 3`] = `
Array [
  920,
  1280,
  1600,
  1920,
  2560,
  3840,
]
`;

exports[`components/atom/SmartImg calculateSizes handles min value correctly 1`] = `
Array [
  120,
  160,
  240,
  320,
  480,
  768,
  920,
  1280,
  1600,
  1920,
  2560,
  3840,
]
`;

exports[`components/atom/SmartImg calculateSizes handles min value correctly 2`] = `
Array [
  1280,
  1600,
  1920,
  2560,
  3840,
]
`;

exports[`components/atom/SmartImg calculateSizes handles min value correctly 3`] = `Array []`;
